### PR TITLE
Implemented generic cmdlets that work with TFS and VSTS

### PR DESCRIPTION
**Pull request by nyanhp**
Added the following new cmdlets (more to come over time):

- Get-TfsBuildDefinition
  - List all build definitions
- Get-TfsBuildDefinitionTemplate
  - List build definition templates
- Get-TfsBuildStep
  - List all available build steps
- New-TfsBuildDefinition
  - Create a new build definition (with n build steps from Get-TfsBuildStep)
- Get-TfsAccessTokenString
  - Get base64 string from the VSTS Personal Access Token
- Get-TfsAgentPool
  - List available agent pools
- Get-TfsProcessTemplate
  - List available process templates (Agile,SCRUM,...)
- Get-TfsProject
  - List available team projects 
- New-TfsProject
  - Create new team project
- Set-TfsProject
  - Change an existing team project
- Get-TfsAgentQueue
  - List all available agent queues
- New-TfsAgentQueue
  - Create a new agent queue
- Get-TfsGitRepository
  - Get a project's git repository

Cmdlets are necessary for AutomatedLab feature TfsApiImplementation.

### CODE REVIEW

**Comment by nyanhp** [comment]
@raandree , can you just have a look if the overall structure makes sense? I will add a sample to our main repository

**Comment by raandree** [comment]
Thanks, the overall structure looks good. I have added some comments.

**Comment by raandree** [comment]
Sorry, forgot to suppress the submit button.

File: AutomatedLab.Common/AutomatedLab.Common.psd1 [hunk 1]
```diff
@@ -58,7 +58,20 @@
...
+        'New-TfsBuildDefinition',
+        'Get-TfsAccessTokenString',
+        'Get-TfsAgentPool',
+        'Get-TfsProcessTemplate',
+        'Get-TfsProject',
+        'New-TfsProject',
+        'Set-TfsProject',
+        'Get-TfsAgentQueue',
+        'New-TfsAgentQueue',
+        'Get-TfsGitRepository'
```
[the remark this answers is not in the document]
Comment (nyanhp) [reply AutomatedLab.Common/AutomatedLab.Common.psd1]: This is automatically generated during the build.

**Comment by nyanhp** [comment]
Harmonized the usage of errors for all Invoke-RestMethod. All cmdlets should generate a terminating error if the action fails.
Reverted back to wildcard for exported cmdlets. Build process will take care of adding all exported functions to the manifest.

**Comment by raandree** [comment]
Shouldn't we rather send non-terminating errors using "Write-Error -Exception"?

**Pull request merged**